packages/cdk/lib/stacks/core-stack.ts

```diff
@@ -96,47 +106,53 @@ export class CoreStack extends Stack {
     new CfnOutput(this, "TableName", { value: this.table.tableName });
   }
 
-  private renderVpc(vpcId?: string): IVpc {
+  private renderVpc(vpcId?: string, publicSubnets?: boolean): IVpc {
     if (vpcId) {
       return Vpc.fromLookup(this, "Vpc", { vpcId });
     }
+
     const vpc = new Vpc(this, "Vpc", {
-      gatewayEndpoints: {
-        S3Endpoint: { service: GatewayVpcEndpointAwsService.S3 },
-      },
+      gatewayEndpoints: publicSubnets
+        ? {}
+        : {
+            S3Endpoint: { service: GatewayVpcEndpointAwsService.S3 },
+          },
+      subnetConfiguration: publicSubnets ? Vpc.DEFAULT_SUBNETS_NO_NAT : Vpc.DEFAULT_SUBNETS,
```

Comment:
Minor point, `Vpc.DEFAULT_SUBNETS_NO_NAT` will still create one private and one public subnet per AZ. The private subnet will be "isolated" from the internet as there is no NAT. If you want to remove private subnets entirely you could have to provide a customized configuration here

Reply:
Do you think there is any advantage in removing the private subnets? I don't think the stack actually puts anything in them, but I don't think we should expose this as an option since it increases the amount of moving parts in the stack, but I would be happy to just change it to only public subnets.

Reply:
I don't think it adds anything related to cost. Can you check if there would be any impact in terms of account level limits on subnets or similar?

Reply:
I agree it won't affect costs. There is a [200-subnet limit on VPCs](https://docs.aws.amazon.com/vpc/latest/userguide/amazon-vpc-limits.html), so I guess this is wasting N subnets, where N is the number of AZs. I guess I might as well change this to only public then.

Reply:
Okay, I've now made it generate only a public subnet.